### ISSUE TITLE
sapwmp should have a way to handle most of the customization

### ISSUE THREAD

**Issue by ricardobranco777** [SUSE/sapwmp#5]
Most of the customization for sapwmp (editing the grub file, profiles, etc.) could be done by the postinstall scriptlet at package installation.

**Comment by nick-wang** [comment]
Good idea! Will add some suitable changes to rpm postinstall.

**Comment by scmschmidt** [comment]
I would be careful here! Let me make some comments about the possible changes.

**Changing the grub entry**
I don't think we can do that safely. If we screw up the boot loader and the system can't boot, we'll have escalations in Support. Also it is not safe to assume, that a customer wants to set up the system immediately with the package installation.
Although a helper script that makes this change including verification and backing up the loader under supervision by the admin, would be very helpful. But it should not be called automatically.

**Altering the profiles**
No automatic configuration here. SAP does not like the idea and I can understand them. One reason is, that there are several ways to configure the profile.  It is possible to import profiles into the database (transaction RZ10) which affects the way on-disk profiles are handled and filesystem profiles get overwritten with the database version and changes in the file are lost.
Again, a script called manually and prints a proper warning and asks for confirmation would be helpful.

**Setting memory low**
For SAP HANA we could at least suggest the value. But the default GAL is not configured anywhere, but compiled in (actually it is a formula). I also haven't found a way yet to read it out. It might be possible, but I suppose the SAP HANA database has to run and we need a SQL statement. Also if SAP changes the formula, we have to keep track.
As long as SAP does not offers an easy interface to determine these parameters, we should not try to guess around. maybe they have and you find it. Would be great!
For SAP Application Servers we can't do anything. There is no configurable limit. Even the SAP developers could not give me a method to calculate or forecast the memory usage. 

Long story short, a "wizard" that helps the admin to set WMP up would be nice to have, but it should be called deliberately and each step has to be verified and confirmed by the user.

**Comment by nick-wang** [comment]
I make sense. We need carefully to do all *automatically* change to avoid any regression.
After reading the comments, yes, maybe add something into rpm phase is not good idea.

We can give some suggestions together with check script and only do change after confirmed. Though i think the possible changes can be done by script are limited.

**Comment by nick-wang** [comment]
The automatically change is risky and increase the burden of maintenance. Considering the SAP may use systemd instead in future, no plan to implement this at the moment.
If there is similar requirement raise in future, more likely in jira instead. So close this issue here.
